### DISCUSSION TITLE
I have an amd gpu and dont know what to do

### DISCUSSION THREAD

**Discussion by amd1890** [open-webui/open-webui#7396] [General]
i am still learning about github and docker

I have a AMD® Ryzen 7 8840u w/ radeon 780m graphics × 16

What is the best way to run this in docker? I am not sure which command to use or what to do.

**Comment by YoussefWaelMohamedLotfy** [comment]
[Quickstart here for Docker](https://docs.openwebui.com/#quick-start-with-docker--recommended)

**Reply by trharmon0** [reply]
That document mentions NVIDIA GPUs only. Would ==--gpus== work for AMD also?

Also, do you know of a similar doc for Docker Compose? I have tried:

    ./run-compose.sh --enable-gpu

but it throws an error about the AMD GPU. The website had some stuff about AMD Radeon and ROCm but only for the TTS plugin.

**Comment by JohnathanCrow** [comment]
I'm on an AMD GPU, and I run:

```
docker run -d -p 3000:8080 --add-host=host.docker.internal:host-gateway -v open-webui:/app/backend/data --name open-webui --restart unless-stopped ghcr.io/open-webui/open-webui:main
```

That's with Ollama already installed via its Windows installer. You might want to change 'unless-stopped' to 'always' if you want to always startup with docker ('unless stopped' will still autostart with docker if you don't manually stop it).

**Reply by amd1890** [reply]
> I'm on an AMD GPU, and I run:
> 
> ```
> docker run -d -p 3000:8080 --add-host=host.docker.internal:host-gateway -v open-webui:/app/backend/data --name open-webui --restart unless-stopped ghcr.io/open-webui/open-webui:main
> ```
> 
> That's with Ollama already installed via its Windows installer. You might want to change 'unless-stopped' to 'always' if you want to always startup with docker ('unless stopped' will still autostart with docker if you don't manually stop it).

windows installer? i use linux

**Reply by JohnathanCrow** [reply]
Yes, but the installation guide doesn't have specific commands for OS. Do you have Ollama installed or are you trying to run the bundled version? I would agree that the guide is a little unclear about the latter.

Edit: Have you tried the command in my post? Does it not work?

**Reply by amd1890** [reply]
> Yes, but the installation guide doesn't have specific commands for OS. Do you have Ollama installed or are you trying to run the bundled version? I would agree that the guide is a little unclear about the latter.
> 
> Edit: Have you tried the command in my post? Does it not work?

I don't remember if I tried the command. I think I did before and it took a long time to download things. I will need to try again sometime.

I don't know the difference between bundled or installed. I would be happy using the bundled one. Ultimately, I am trying to get Perplexica to work using ollama utilizing the AMD GPU. When I run GPT4ALL, I can use the GPU and it's moderately fast. When I use Perplexica in docker, it's terribly slow and keeps using CPU.

I came across this program as a great interface for Ollama and would like to use the interface for Ollama but using the GPU, and then I am pretty sure if it gives me a port and it's in docker, I can somehow link it to Perplexica.

This is so hard for me. I am not as smart as many of the other people on github but still keep trying to learn things. I'll try again soon to use the command. I do not have fast internet and it's many gigs of files, if i recall right.

**Reply by JohnathanCrow** [reply]
The bundled version installs both Ollama and Open WebUI via docker. The other option is to have Ollama installed on your computer (via its own installer), and Open WebUI in docker.

So do you already have Ollama installed using the Ollama installer? Or is it running in docker?

**Reply by amd1890** [reply]
I have both installed, although right now, when I open whaler, everything is greyed out. it looks like i have ollama/ollama and openwebui and perplexica running, but I have no idea why everything is greyed out or even if any of them are running.

**Reply by JohnathanCrow** [reply]
It's a little hard for me to help because I run on Windows. Is your GPU on this list?

![{436F9C4B-54F0-4BB8-BC4A-DB9D3124F448}](https://github.com/user-attachments/assets/741711da-66c2-4e02-a3ae-05239e0d40f2)

Also, this link might be of use: https://github.com/ollama/ollama/blob/main/docs/linux.md

Hopefully someone a little more knowledgeable will happen by this thread.

**Reply by dingoliberty** [reply]
> This is so hard for me. I am not as smart as many of the other people on github but still keep trying to learn things. I'll try again soon to use the command. I do not have fast internet and it's many gigs of files, if i recall right.

dont be so down on yourself dude

**Comment by NLxAROSA** [comment]
The default ROCm libraries don't support the 780m iGPU. They can be modded and added though, but the Ollama team has not implemented that in their versions, so for now everyone has to do that themselves until they update it.

Luckily, someone made a fork with the 780m support added in by default: https://github.com/likelovewant/ollama-for-amd 

I run it on my ROG Ally handheld using that. :)

I also believe some folks made it into a container, so if you replace the default ollama tag with the one that has the correct ROCm libs that should also work.
